src/c/modularizace/automatizace_prekladu.md

```diff
@@ -0,0 +1,75 @@
+pomocí kterého můžeme všechny tyto úkony provést pomocí jediného příkazu v terminálu, až po pokročilé
+**sestavovací systémy** (*build systems*), které umí automaticky vyhledávat cesty ke knihovnám a
+překládat pouze změněné soubory pro urychlení opakovaných překladů programu.
+
+Bohužel neexistuje jednotný standardní sestavovací systém pro programy napsané v *C*. Různé projekty
+či knihovny tak používají různé sestavovací systémy, což může někdy představovat problém při jejich
+integraci do našich programů. Sestavovací systémy se navíc obvykle nastavují pomocí konfiguračních
+souborů, které jsou psány v proprietárních jazycích, které se musíte naučit a pochopit, abyste daný
+sestavovací systém mohli používat. Situaci nepomáhá ani to, že se od sebe jednotlivé systémy značně
+liší a bývají velmi složité. 
+
+## [`make`](https://en.wikipedia.org/wiki/Make_(software))
+Asi stále nejpoužívanějším sestavovacím systémem je `make`, který existuje už od roku 1976. Pro jeho
+použití musíte vytvořit soubor `Makefile`, ve kterém popíšete, jak se má váš program přeložit, a poté
+spustíte program `make`, který jej dle konfiguračního souboru přeloží.
+  
+Návod pro vytvoření konfiguračního souboru `Makefile` a použití `make` naleznete například
+[zde](https://www.itnetwork.cz/cecko/linux/tutorial-c-linux-makefile).
+
+## [`CMake`](https://cmake.org/)
+Poněkud modernější alternativou je `CMake`. Jedná se o meta systém, ve skutečnosti totiž neřídí překlad
+vašeho programu, ale pouze generuje potřebné soubory pro nějaký jiný sestavovací systém, který váš
+program teprve přeloží. Výhodou pak je, že z jednoho `CMake` konfiguračního souboru tak můžete vygenerovat
+např. `Makefile` pro přeložení na Linuxu anebo jiné konfigurační soubory pro přeložení stejného programu
+pod Windows. Další výhodou `CMake` je, že některá vývojová prostředí (např.
+[Visual Studio Code](https://code.visualstudio.com/) nebo [CLion](https://www.jetbrains.com/clion/))
+mu rozumí a dokáží díky němu usnadnit analýzu a ladění vašeho programu. 
+
+Pro použití `CMake` musíte vytvořit konfigurační soubor `CMakeLists.txt`, ve kterém popíšete jednotlivé
+zdrojové soubory vašeho programu, a také zadáte knihovny, které chcete k vašemu programu připojit.
+Vzorový soubor `CMakeLists.txt` může vypadat např. takto:
+```cmake
+cmake_minimum_required(VERSION 3.4)
+
+# Název projektu
+project(hra)
+
+# Přidání přepínačů překladače
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=address")
+
+# Vyhledání knihovny SDL
+find_package(SDL2)
+
+# Vytvoření programu s názvem `hra`
+# Program se bude skládat ze dvou zadaných zdrojových souborů (jednotek překladu)
+add_executable(hra hra.c grafika.c)
+
+# Přidání adresářů s hlavičkovými soubory k programu (obdoba -I)
+target_include_directories(hra PRIVATE ${SDL2_INCLUDE_DIRS})
+
+# Přilinkování knihoven k programu (obdoba -l)
+target_link_libraries(hra ${SDL2_LIBRARIES} m)
+```
```

Comment:
Myslim ,ze je ten priklad zabije, ale OK. Ti co chteji tak se neco nauci. Zbytek bude neprecte nebo zapomene.

Reply:
Jasně, nemám chuť vysvětlovat CMake ani Makefile ve skriptech. Je to tam řečeno jenom ať ví, že to existuje.